### PR TITLE
Added new tests to support PDF downloads of lists and PDF downloads of summary pages for 5.7 only

### PR DESCRIPTION
**Pull request by hhovsepy**
Created random_methods.py file and moved all common random objet get methods into it.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 3d5c6a555a4db92dad1628722971b9835888607c:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/middleware/__init__.py [hunk 1]
```diff
@@ -32,6 +33,9 @@ def _on_detail_page(self):
         ensure_browser_open()
         return sel.is_displayed('//h1[contains(., "{} (Summary)")]'.format(self.name))
 
+    def download_summary(self):
+        self.summary.reload()
+        download_summary()
```
Comment (psav) [inline cfme/middleware/__init__.py]: Why not just call the ```download_summary_btn()``` method?

File: cfme/tests/middleware/test_middleware_downloads.py [hunk 2]
```diff
@@ -1,85 +1,90 @@
...
+FILETYPES = ["txt", "csv", "pdf"]
 
 
-@pytest.mark.parametrize("filetype", ["txt", "csv"])
+@pytest.mark.parametrize("filetype", FILETYPES)
 @pytest.mark.parametrize("objecttype", [MiddlewareDatasource, MiddlewareDeployment,
                                         HawkularProvider, MiddlewareServer,
                                         MiddlewareDomain, MiddlewareMessaging])
+@pytest.mark.uncollectif(lambda filetype: filetype in {"pdf"} and
+                         current_version() == version.UPSTREAM)
```
Comment (psav) [inline cfme/tests/middleware/test_middleware_downloads.py]: PDF unavailable in upstream?

File: cfme/middleware/__init__.py (hunk 1 above)
Comment (hhovsepy) [inline cfme/middleware/__init__.py]: agree, unnecessary splitting, as it is for all middleware objects, thanks

**Comment by dajoRH** [comment]
Lint report for commit 76ba16eeb92082cd4db4b4de660a77391bde51cd:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 93ee310dd2e086f5676caf1b4d6b2779271cd13b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 98ab54351a203ce26d75fbcafb1345b6370c1f17:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e791c26b616bba45204e24ae7ccd7ad2548eaee7:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**